tests/planner/kinodynamic/test_KinodynamicTimer.cpp

```diff
@@ -0,0 +1,392 @@
+using aikido::statespace::R2;
+using aikido::statespace::CartesianProduct;
+using aikido::statespace::SO2;
+using aikido::statespace::StateSpacePtr;
+using aikido::statespace::ConstStateSpacePtr;
+using aikido::planner::kinodynamic::computeKinodynamicTiming;
+using aikido::planner::parabolic::convertToSpline;
+
+class KinodynamicTimerTests : public ::testing::Test
+{
+public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+protected:
+  void SetUp() override
+  {
+    mStateSpace = std::make_shared<R2>();
+    mMaxVelocity = Eigen::Vector2d(1., 1.);
+    mMaxAcceleration = Eigen::Vector2d(2., 2.);
+
+    mInterpolator = std::make_shared<GeodesicInterpolator>(mStateSpace);
+    mStraightLine = std::make_shared<Interpolated>(mStateSpace, mInterpolator);
+
+    auto state = mStateSpace->createState();
+
+    state.setValue(Vector2d(1., 2.));
+    mStraightLine->addWaypoint(0., state);
+
+    state.setValue(Vector2d(3., 4.));
+    mStraightLine->addWaypoint(1., state);
+  }
+
+  std::shared_ptr<R2> mStateSpace;
+  Eigen::Vector2d mMaxVelocity;
+  Eigen::Vector2d mMaxAcceleration;
+
+  std::shared_ptr<GeodesicInterpolator> mInterpolator;
+  std::shared_ptr<Interpolated> mStraightLine;
+};
+
+TEST_F(KinodynamicTimerTests, MaxVelocityIsZero_Throws)
+{
+  Vector2d zeroMaxVelocity(1., 0.);
+  EXPECT_THROW(
+      {
+        computeKinodynamicTiming(
+            *mStraightLine, zeroMaxVelocity, mMaxAcceleration);
+      },
+      std::invalid_argument);
+}
+
+TEST_F(KinodynamicTimerTests, MaxVelocityIsNegative_Throws)
+{
+  Vector2d negativeMaxVelocity(1., -1);
+  EXPECT_THROW(
+      {
+        computeKinodynamicTiming(
+            *mStraightLine, negativeMaxVelocity, mMaxAcceleration);
+      },
+      std::invalid_argument);
+}
+
+TEST_F(KinodynamicTimerTests, MaxAccelerationIsZero_Throws)
+{
+  Vector2d zeroMaxAcceleration(1., 0.);
+  EXPECT_THROW(
+      {
+        computeKinodynamicTiming(
+            *mStraightLine, mMaxVelocity, zeroMaxAcceleration);
+      },
+      std::invalid_argument);
+}
+
+TEST_F(KinodynamicTimerTests, MaxAccelerationIsNegative_Throws)
+{
+  Vector2d negativeMaxAcceleration(1., -1);
+  EXPECT_THROW(
+      {
+        computeKinodynamicTiming(
+            *mStraightLine, mMaxVelocity, negativeMaxAcceleration);
+      },
+      std::invalid_argument);
+}
+
+TEST_F(KinodynamicTimerTests, StartsAtNonZeroTime)
+{
+  Interpolated inputTrajectory(mStateSpace, mInterpolator);
+
+  auto state = mStateSpace->createState();
+
+  // This is the same test as StraightLine_TriangularProfile, except that the
+  // trajectory starts at a non-zero time.
+  state.setValue(Vector2d(1., 2.));
+  inputTrajectory.addWaypoint(1., state);
+
+  state.setValue(Vector2d(2., 3.));
+  inputTrajectory.addWaypoint(3., state);
+
+  auto timedTrajectory = computeKinodynamicTiming(
+      inputTrajectory, Vector2d::Constant(2.), Vector2d::Constant(1.));
+
+  timedTrajectory->evaluate(1., state);
+  EXPECT_TRUE(Vector2d(1.0, 2.0).isApprox(state.getValue()));
```

Comment:
I think there are some Eigen things in https://github.com/personalrobotics/aikido/blob/master/tests/eigen_tests.hpp that we could use instead? The tests would be more descriptive when they fail.